R/benchTradePerf.R

```diff
@@ -0,0 +1,200 @@
+#' Trade Execution Performance Benchmarks
+#' 
+#' This function gathers different benchmarking methods used to evaluate the average execution price \eqn{\bar{P}} of a given trading strategy.
+#' The \eqn{\bar{P}} is compared against a number of benchmark metrics, in order to assess its performance in terms of profit or loss.
+#' These methods are not strictly alternative, indeed each of them provides different insights and suffers limitations.
+#' 
+#' The performance is quantified by means of a \emph{Profit and Loss (PnL) metric}. 
+#' A positive PnL metric indicates that the trading strategy outperformed a chosen benchmark on average,
+#' viceversa negative values register an underperformance.
+#' 
+#' By and large, PnL metrics are computed as:
+#' 
+#' 
+#' where \eqn{P_{avg}} is the average execution price and \eqn{P_{B}} is a given benchmark price that can be a single next/present open/close price.
+#' It is worth stressing that they are expressed in basis points (bps) units.
+#' 
+#' A widely used one is the \emph{Volume Weighted Average Price (VWAP) benchamark}... 
+#' 
+#' A variation of the VWAP benchmark is given by the \emph{Participation Weighted Price (PWP) benchmark}.
+#' 
+#' 
+#' Lastly the \emph{Relative Performance Measure}, which differs from the PnL metrics above. It is expressed as:
+#' 
+#' 
+#' 
+#' @param Portfolio A portfolio name that points to a portfolio object structured with initPortf()
+#' @param Symbol A string identifying the traded symbol to benchmark
+#' @param side Numeric, either 1 or -1, that indicates the side of the trade. \code{side = 1} means "Buy" and \code{side = -1} is "Sell". Default is \code{side = 1}.
+#' @param benchmark A string or vector of strings providing the benchmark metric(s). One of 'MktBench', 'VWAP', 'PWP', 'RPM'. Default is #TODO.
+#' @param type A string specifying the type of the benchmark used. Relevant only for \code{benchmark = 'MktBench'} and \code{benchmark = 'VWAP'}. 
+#'             In the former case it only changes the output format, while in the latter it also specifies the VWAP to be used. If \code{benchmark = 'VWAP'}, default is \code{type = 'Txns'}
+#' @param MktData An xts object containing a 'MktPrice' and 'MktVolmn' required columns.
```

Comment:
Indeed I think `MktData` is consistent with `TxnData` in `addTxns`, for example. 

I didn't like much `Volmn` either. If we agree it will be turned in `Qty` all around.

Reply:
So keeping `MktData` param and going ahead with `MktQty` edits?
It overall seems to integrate well with corresponding transactions naming conventions of the package.

Reply:
agree with Qty